src/mono/browser/runtime/jiterpreter-trace-generator.ts

```diff
@@ -3953,6 +3954,12 @@ function emit_simd_4 (builder: WasmBuilder, ip: MintOpcodePtr, index: SimdIntrin
 function emit_atomics (
     builder: WasmBuilder, ip: MintOpcodePtr, opcode: number
 ) {
+    if (opcode === MintOpcode.MINT_MONO_MEMORY_BARRIER) {
+        if (WasmEnableThreads)
+            builder.callImport("membar");
+        return true;
+    }
+
```

Comment:
Why? `mono_memory_barrier` is just `__sync_synchronize` (it's an ABI memory barrier, not a GC write barrier), so you could just emit `atomic.fence` directly, no?

https://godbolt.org/z/jjWsxzj9h